src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -4724,6 +4731,16 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
 
 	rte->eref = alias;
 
+	/*
+		Store the alias information only when table alias is used to allow 
+		EXPLAIN find it
+	*/
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		rte->alias = (Alias *) copyObject(alias);
```

Comment:
Why both `rte->eref` and `rte->alias` are fileld with the same value?

Reply:
> Why both `rte->eref` and `rte->alias` are filled with the same value? And why we are making copy for `rte->alias`, but not for `rte->eref`?

https://github.com/arenadata/gpdb/pull/780#discussion_r1495261708

Reply:
Yes, in this particular case they have equal value. But the fields has different purpose and potentially different life cycle, that's why value is copied.